components/ResultsPage/WillBeEligible.tsx

```diff
@@ -30,7 +33,29 @@ export const WillBeEligible: React.VFC<{
 
       {futureResults.map((resultObj, idx) => {
         const age = Object.keys(resultObj)[0]
-        const text = `${language === 'en' ? 'At' : 'À'} ${age}, ${
+        const onlyOASGIS = Object.keys(
+          resultObj[Object.keys(resultObj)[0]]
+        ).filter((key) => key === 'oas' || key === 'gis')
+
+        // show if some are non zero
+        const nonZeroExist = onlyOASGIS.some(
+          (key) => resultObj[age][key].entitlement?.result > 0
+        )
+
+        const enStr =
+          multipleOAS_GIS && nonZeroExist
+            ? partner
+              ? 'If your partner starts receiving at'
```

Comment:
TODO  adding this ones to the messages

Reply:
Agreed! IF the decision is to keep things this way. But I think this is meant to be a temporary fix until we figure out a better way to display multiple OAS/GIS benefits or start accounting for that "years in Canada" thing that we talked about